ext/php7/ddtrace.c

```diff
@@ -450,12 +450,13 @@ static PHP_MINIT_FUNCTION(ddtrace) {
      * See http://www.phpinternalsbook.com/php7/extensions_design/zend_extensions.html#hybrid-extensions
      * {{{ */
     zend_register_extension(&_dd_zend_extension_entry, ddtrace_module_entry.handle);
-#ifdef COMPILE_DL_DDTRACE
-    Dl_info infos;
-    // The symbol used needs to be public on Alpine.
-    dladdr(get_module, &infos);
-    dlopen(infos.dli_fname, RTLD_LAZY);
-#endif
+    zend_module_entry *mod_ptr =
+        zend_hash_str_find_ptr(&module_registry, PHP_DDTRACE_EXTNAME, sizeof(PHP_DDTRACE_EXTNAME) - 1);
+    if (mod_ptr == NULL) {
+        // This shouldn't happen, possibly a bug if it does.
```

Comment:
This should invoke zend_error() then (core warning?), with a message similar to "Did not find ddtrace extension in registered modules. Please open a bug report."